base/bootstrap.go

```diff
@@ -78,6 +88,25 @@ func (cc *CouchbaseCluster) connect() (*gocb.Cluster, error) {
 	return cluster, nil
 }
 
+// connect attempts to open a gocb.Cluster connection. Callers will be responsible for closing the connection.
+func (cc *CouchbaseCluster) connect() (*gocb.Cluster, error) {
```

Comment:
We don't have that many places that call connect().  Did you consider just adding authenticator to the connect() function.  I think that might make it more obvious what callers are doing (i.e. they are either passing in the cluster credentials, or bucket credentials).